noah/src/main/scala/edu/uci/ics/cloudberry/noah/feed/TwitterFeedStreamDriver.java

```diff
@@ -80,11 +82,15 @@ public static void run(Config config)
             isConnected = true;
             // Do whatever needs to be done with messages
             while (true) {
-                String msg = queue.take();
-                String adm = tagTweet.tagOneTweet(msg) + "\n";
-                if ((adm.length() > 0) && (!adm.equals("\n"))) {
+                try {
+                    String msg = queue.take();
+                    String adm = tagTweet.tagOneTweet(msg);
                     socketAdapterClient.ingest(adm);
-                    bw.write(adm);
+                    bw.write(msg);
```

Comment:
Plus, the number of records from twitter is huge. Could you do some research to make it write a gzip file directly?